packages/palette/src/themes/Themes.story.tsx

```diff
@@ -0,0 +1,224 @@
+        other.
+      </Text>
+
+      <Separator color="black30" my={12} />
+
+      <GridColumns>
+        <Column span={6}>
+          <Text variant="sm" color="black60">
+            Color Value
+          </Text>
+        </Column>
+
+        <Column span={6}>
+          <Text variant="sm" color="black60">
+            Hex Value
+          </Text>
+        </Column>
+      </GridColumns>
+
+      <Separator color="black30" my={6} />
+
+      {colors.map(([name, value]) => {
+        return (
+          <GridColumns key={name} justifyContent="space-between" my={4}>
+            <Column span={6} display="flex" alignItems="center">
+              <Box width={60} height={60} bg={name} borderRadius="50%" />
+              <Text variant="xl" ml={2}>
+                {name}
+              </Text>
+            </Column>
+
+            <Column span={6} display="flex" alignItems="center">
+              <Text variant="sm" color="black60">
+                color:&nbsp;
+              </Text>
+              <Text variant="sm">{value}</Text>
+            </Column>
+          </GridColumns>
+        )
+      })}
+    </Box>
+  )
+}
+
+export const Spacing = () => {
+  const { theme } = useTheme()
+
+  const spacing = Object.keys(theme.space).sort(
+    (a, b) => parseFloat(a) - parseFloat(b)
+  )
+
+  return (
+    <Box>
+      <Text variant="xxl" mb={4}>
+        Spacing
+      </Text>
+
+      <Join separator={<Spacer my={2} />}>
+        {spacing.map((key) => {
+          const px = parseFloat(key) * 10
+
+          return (
+            <Box key={key}>
+              <Text variant="lg">{key}</Text>
+              <Text variant="xs">{px}px</Text>
+              <Box width={px} height={1} bg="black60" />
+            </Box>
+          )
+        })}
+      </Join>
+    </Box>
+  )
+}
+
+export const Grid = () => {
+  const { theme } = useTheme()
+
+  return (
+    <Join separator={<Spacer my={2} />}>
+      <Text variant="xxl">Desktop</Text>
+
+      <GridColumns my={12} width="100%" height={800}>
+        {[...new Array(12)].map((_, i) => (
+          <Column key={i} span={[1]} bg="black10" height="100%" />
+        ))}
+      </GridColumns>
+
+      <Separator my={12} color="black30" />
+
+      <Text variant="xxl">Tablet</Text>
+
+      <GridColumns my={12} width={theme.breakpoints.sm} height={800} mx="auto">
+        {[...new Array(12)].map((_, i) => (
+          <Column key={i} span={[1]} bg="black10" height="100%" />
+        ))}
+      </GridColumns>
+
+      <Separator my={12} color="black30" />
+
+      <Text variant="xxl">Mobile</Text>
+
+      <GridColumns my={12} width={480} gridColumnGap={1} height={800} mx="auto">
+        {[...new Array(12)].map((_, i) => (
+          <Column key={i} span={[1]} bg="black10" height="100%" />
+        ))}
+      </GridColumns>
+    </Join>
+  )
+}
+
+export const Typography = () => {
+  const { theme } = useTheme()
+
+  const variants =
+    theme.id === "v2" ? theme.textVariants.large : theme.textVariants
```

Comment:
Switching on the id like this gives you back the type safe theme. You can also pass a theme type: `useTheme<ThemeV3>()` if you know you want a specific one.